pool/hub.go

```diff
@@ -259,18 +259,33 @@ func NewHub(cancel context.CancelFunc, hcfg *HubConfig) (*Hub, error) {
 	}
 	h.chainState = NewChainState(sCfg)
 
-	mode := uint32(0)
+	// Ensure database is in the correct mode.
+	dbMode, err := hcfg.DB.fetchSoloPoolMode()
+	if err != nil {
+		// If pool mode is not set, assume this is a new database and insert
+		// pool mode now.
```

Comment:
s / `... assume this is a new database and insert pool mode now` / `...assume the database is new and persist the pool mode.`